test/unit/storage/storage.spec.ts

```diff
@@ -29,88 +29,110 @@ describe('Storage', () => {
+        // We expect referential equality here
+        expect(storage.app).to.equal(mockApp);
+      });
+  
+      it('is read-only', () => {
+        expect(() => {
+          (storage as any).app = mockApp;
+        }).to.throw('Cannot set property app of #<Storage> which has only a getter');
+      });
     });
-  });
-
-  describe('app', () => {
-    it('returns the app from the constructor', () => {
-      // We expect referential equality here
-      expect(storage.app).to.equal(mockApp);
+  
+    describe('bucket(invalid)', () => {
+      const expectedError = 'Bucket name not specified or invalid. Specify a valid bucket name via ' +
+          'the storageBucket option when initializing the app, or specify the bucket name ' +
+          'explicitly when calling the getBucket() method.';
+      const invalidNames = [null, NaN, 0, 1, true, false, '', [], [1, 'a'], {}, { a: 1 }, _.noop];
+      invalidNames.forEach((invalidName) => {
+        it(`should throw given invalid bucket name: ${ JSON.stringify(invalidName) }`, () => {
+          expect(() => {
+            const bucketAny: any = storage.bucket;
+            bucketAny(invalidName);
+          }).to.throw(expectedError);
+        });
+      });
     });
-
-    it('is read-only', () => {
-      expect(() => {
-        (storage as any).app = mockApp;
-      }).to.throw('Cannot set property app of #<Storage> which has only a getter');
+  
+    describe('bucket()', () => {
+      it('should return a bucket object', () => {
+        expect(storage.bucket().name).to.equal('bucketName.appspot.com');
+      });
     });
-  });
-
-  describe('bucket(invalid)', () => {
-    const expectedError = 'Bucket name not specified or invalid. Specify a valid bucket name via ' +
-        'the storageBucket option when initializing the app, or specify the bucket name ' +
-        'explicitly when calling the getBucket() method.';
-    const invalidNames = [null, NaN, 0, 1, true, false, '', [], [1, 'a'], {}, { a: 1 }, _.noop];
-    invalidNames.forEach((invalidName) => {
-      it(`should throw given invalid bucket name: ${ JSON.stringify(invalidName) }`, () => {
-        expect(() => {
-          const bucketAny: any = storage.bucket;
-          bucketAny(invalidName);
-        }).to.throw(expectedError);
+  
+    describe('bucket(valid)', () => {
+      it('should return a bucket object', () => {
+        expect(storage.bucket('foo').name).to.equal('foo');
       });
     });
   });
 
-  describe('bucket()', () => {
-    it('should return a bucket object', () => {
-      expect(storage.bucket().name).to.equal('bucketName.appspot.com');
-    });
-  });
+  describe('Emulator Support', () => {
+    it('sets STORAGE_EMULATOR_HOST if FIREBASE_STORAGE_EMULATOR_HOST is set', () => {
+      const EMULATOR_HOST = 'http://localhost:9199';
+      delete process.env.STORAGE_EMULATOR_HOST;
+      process.env.FIREBASE_STORAGE_EMULATOR_HOST = EMULATOR_HOST;
+
+      mockApp = mocks.app();
```

Comment:
Already initialized in a `beforeEach` hook above.

Reply:
The `beforeEach` hook is now only for the non-emulator tests. I had to split it into 2 because I need to set the env variable before the `beforeEach` hook starts.

Reply:
Done.